control/src/main.rs

```diff
@@ -0,0 +1,162 @@
+mod runtime;
+mod commands;
+mod helpers;
+
+use codec::Encode;
+use clap::{Parser, Subcommand, ValueEnum};
+use subxt::{OnlineClient, PolkadotConfig};
+use std::io::Write;
+use alloy_primitives::{Address, Bytes, FixedBytes};
+
+#[derive(Debug, Parser)]
+#[command(name = "snowbridge-control", version, about, long_about = None)]
+struct Cli {
+    /// Output format of preimage
+    #[arg(value_enum, default_value_t=Format::Hex)]
+    format: Format,
+
+    #[command(subcommand)]
+    command: Command,
+}
+
+#[derive(Debug, Subcommand)]
+pub enum Command {
+    /// Change the gateway operating mode
+    GatewayOperatingMode {
+        /// Operating mode
+        #[arg(value_enum)]
+        mode: GatewayOperatingModeArg,
+    },
```

Comment:
Having the set operating mode command is useful but this would also need to be in the same pre-image as the `Upgrade` otherwise an upgrade would require two proposals. 
The pre-image for upgrade needs to:
1. Set to mode `RejectingOutbound`.
2. Call upgrade extrinsic.
3. Set mode to `Normal`

Reply:
I designed Gateway upgrades to not need parts of the bridge to be disabled first. At least for most routine upgrades.